Source/ImmutableMarketplace/Private/ImmutableMarketplaceLinkFactory.cpp

```diff
@@ -0,0 +1,104 @@
+﻿#include "ImmutableMarketplaceLinkFactory.h"
+
+#include "PlatformHttp.h"
+
+#include "ImmutableMarketplaceEnums.h"
+#include "ImmutableMarketplaceLinkConfig.h"
+
+FString UImmutableMarketplaceLinkFactory::GenerateOnRampLink(EImmutableEnvironment Environment, FString Email, FString WalletAddress, const FImmutableOnRampQueryParams& QueryParams, TMap<FString, FString> ExtraQueryParams)
+{
+	auto FindExtraQueryParam = [&](const FString& Key, const FString& DefaultValue) -> FString
+	{
+		if (FString* Value = ExtraQueryParams.Find(Key))
+		{
+			return *Value;
+		}
+
+		return DefaultValue;
+	};
+
+	const FString BaseUrl = UImmutableMarketplaceLinkConfig::GetBaseUrl(Environment, EImmutableMarketplaceFlow::OnRamp);
+	const FString ApiKey = UImmutableMarketplaceLinkConfig::GetApiKey(Environment, EImmutableMarketplaceFlow::OnRamp);
+
+	TArray<FString> QueryParamsArray
+	{
+		FString::Printf(TEXT("%s=%s"), TEXT("apiKey"), *FPlatformHttp::UrlEncode(ApiKey)),
+		FString::Printf(TEXT("%s=%s"), TEXT("cryptoCurrencyList"), *FPlatformHttp::UrlEncode(QueryParams.CryptoCurrencyList)),
+		FString::Printf(TEXT("%s=%s"), TEXT("defaultCryptoCurrency"), *FPlatformHttp::UrlEncode(QueryParams.DefaultCryptoCurrency)),
+		FString::Printf(TEXT("%s=%s"), TEXT("defaultFiatAmount"), *FPlatformHttp::UrlEncode(QueryParams.DefaultFiatAmount)),
+		FString::Printf(TEXT("%s=%s"), TEXT("defaultFiatCurrency"), *FPlatformHttp::UrlEncode(QueryParams.DefaultFiatCurrency)),
+		FString::Printf(TEXT("%s=%s"), TEXT("defaultPaymentMethod"), *FPlatformHttp::UrlEncode(FindExtraQueryParam(TEXT("defaultPaymentMethod"), TEXT("credit_debit_card")))),
+		FString::Printf(TEXT("%s=%s"), TEXT("disablePaymentMethods"), *FPlatformHttp::UrlEncode(FindExtraQueryParam("disablePaymentMethods", ""))),
+		FString::Printf(TEXT("%s=%s"), TEXT("disableWalletAddressForm"), *FPlatformHttp::UrlEncode(FindExtraQueryParam(TEXT("disableWalletAddressForm"), TEXT("true")))),
+		FString::Printf(TEXT("%s=%s"), TEXT("email"), *FPlatformHttp::UrlEncode(Email)),
+		FString::Printf(TEXT("%s=%s"), TEXT("exchangeScreenTitle"), *FPlatformHttp::UrlEncode(FindExtraQueryParam("exchangeScreenTitle", TEXT("Buy")))),
+		FString::Printf(TEXT("%s=%s"), TEXT("isAutoFillUserData"), *FPlatformHttp::UrlEncode(FindExtraQueryParam(TEXT("isAutoFillUserData"), TEXT("true")))),
+		FString::Printf(TEXT("%s=%s"), TEXT("network"), TEXT("immutablezkevm")),
+		FString::Printf(TEXT("%s=%s"), TEXT("productsAvailed"), TEXT("buy")),
+		FString::Printf(TEXT("%s=%s"), TEXT("themeColor"), *FPlatformHttp::UrlEncode(FindExtraQueryParam(TEXT("themeColor"), TEXT("0D0D0D")))),
+		FString::Printf(TEXT("%s=%s"), TEXT("walletAddress"), *FPlatformHttp::UrlEncode(WalletAddress))
+	};
```

Comment:
Once this is done, don't you need to add any extra query parameters from the `ExtraQueryParams` to the `QueryParamsArray`?  

For example, if `ExtraQueryParams` includes `hideMenu`, which isn't in your list but is a valid Transak query parameter ([reference](https://docs.transak.com/docs/query-parameters)), you should include it?